### PR TITLE
Serialization for scalar functions nextval and currval

### PR DESCRIPTION
**Pull request by ywelsch**
Adds serialization / deserialization code for the scalar functions `nextval` and `currval` that are used by DuckDB's sequences feature. The PR also moves the `NextvalBindData` struct to the header file so that it can be used by extensions at planning time.

### CODE REVIEW

File: src/include/duckdb/function/scalar/sequence_functions.hpp [hunk 1]
```diff
@@ -8,12 +8,34 @@
...
+struct NextvalBindData : public FunctionData {
+	explicit NextvalBindData(optional_ptr<SequenceCatalogEntry> sequence)
+	    : sequence(sequence), create_info(sequence ? sequence->GetInfo() : nullptr) {
+	}
+
+	//! The sequence to use for the nextval computation; only if the sequence is a constant
+	optional_ptr<SequenceCatalogEntry> sequence;
+
+	//! The CreateInfo for the above sequence, if it exists
+	unique_ptr<CreateInfo> create_info;
```
Comment (Mytherin) [inline src/include/duckdb/function/scalar/sequence_functions.hpp:27]: Is this property necessary here? Can't we instead fetch this when `Serialize` is called?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks good to me - it's only a bit unclear why the `CreateInfo` is stored in the bind data when it is only really used as an intermediate during serialization. Perhaps it would be cleaner to directly serialize the catalog/schema/sequence name instead of serializing the `CreateInfo` as well?

File: src/include/duckdb/function/scalar/sequence_functions.hpp (hunk 1 above)
Comment (ywelsch) [reply src/include/duckdb/function/scalar/sequence_functions.hpp:27]: We currently rely on the ability to override the catalog name in the create_info before sending it over the wire (we do a similar thing to the info in LogicalCreateTable / LogicalSimple / LogicalCreate). The reason is that the virtual catalog on the client-side is using a different catalog name than what's used on the server-side (globally unique FQN names for the DBs), and rebinding happens on deserialization.

I understand that it looks a bit odd, but I haven't found a better way to do this.

**Comment by Mytherin** [comment]
Thanks for clarifying, LGTM

**Pull request merged**